Userland/Utilities/pdf.cpp

```diff
@@ -16,34 +16,62 @@
 #include <LibPDF/Document.h>
 #include <LibPDF/Renderer.h>
 
-static PDF::PDFErrorOr<void> print_document_info_dict(PDF::Document& document)
+static PDF::PDFErrorOr<void> print_document_info(PDF::Document& document, bool json)
 {
+    JsonObject json_output;
+
+    if (!json) {
+        outln("PDF Version: {}.{}", document.version().major, document.version().minor);
+        outln("Number of pages: {}", document.get_page_count());
+    } else {
+        JsonObject version;
+        version.set("major", document.version().major);
+        version.set("minor", document.version().minor);
+        json_output.set("version", version);
+        json_output.set("page_count", document.get_page_count());
+    }
+
     if (auto info_dict = TRY(document.info_dict()); info_dict.has_value()) {
-        if (auto title = TRY(info_dict->title()); title.has_value())
-            outln("Title: {}", title);
-        if (auto author = TRY(info_dict->author()); author.has_value())
-            outln("Author: {}", author);
-        if (auto subject = TRY(info_dict->subject()); subject.has_value())
-            outln("Subject: {}", subject);
-        if (auto keywords = TRY(info_dict->keywords()); keywords.has_value())
-            outln("Keywords: {}", keywords);
-        if (auto creator = TRY(info_dict->creator()); creator.has_value())
-            outln("Creator: {}", creator);
-        if (auto producer = TRY(info_dict->producer()); producer.has_value())
-            outln("Producer: {}", producer);
-        if (auto creation_date = TRY(info_dict->creation_date()); creation_date.has_value())
-            outln("Creation date: {}", creation_date);
-        if (auto modification_date = TRY(info_dict->modification_date()); modification_date.has_value())
-            outln("Modification date: {}", modification_date);
+        if (!json) {
+            if (auto title = TRY(info_dict->title()); title.has_value())
+                outln("Title: {}", title);
```

Comment:
Can you come up with a way of writing this that doesn't duplicate the function? Maybe add a local `emit(long_text, json_key, T value)` lambda that either outln()s or set()s based on `json` and call `emit()` instead of `outln()` or something.

Reply:
Done!